packages/jest-haste-map/src/lib/dependencyExtractor.ts

```diff
@@ -54,7 +54,7 @@ const REQUIRE_OR_DYNAMIC_IMPORT_RE = createRegExp(
 
 const IMPORT_OR_EXPORT_RE = createRegExp(
   [
-    '\\b(?:import|export)\\s+(?!type(?:of)?\\s+)[^\'"]+\\s+from\\s+',
+    '\\b(?:import|export)\\s+(?!type(?:of)?\\s+)(?:[^\'"]+\\s+from\\s+)?',
```

Comment:
I know what `(?:...)` is doing, but why is the `?` at the end necessary?

Reply:
The `?` at the end makes the entire group optional.  Side-effect only imports have no ` from ` text.